packages/cryptoplease/lib/features/outgoing_transfer/presentation/send_flow/fungible_token/link_transfer_flow_screen.dart

```diff
@@ -29,16 +32,33 @@ class FtLinkTransferFlowScreen extends StatefulWidget {
 }
 
 class _FtLinkTransferFlowScreenState extends State<FtLinkTransferFlowScreen> {
+  late final FtCreateOutgoingTransferBloc outgoingBloc;
+
+  @override
+  void initState() {
+    super.initState();
+    final amount = widget.amount;
+    final token = amount?.token ?? widget.token;
+
+    outgoingBloc = FtCreateOutgoingTransferBloc(
+      repository: context.read<OutgoingTransferRepository>(),
+      balances: context.read<BalancesBloc>().state.balances,
+      conversionRatesRepository: context.read<ConversionRatesRepository>(),
+      userCurrency: context.read<UserPreferences>().fiatCurrency,
+      transferType: OutgoingTransferType.splitKey,
+      initialToken: token,
+    );
+
+    if (amount != null) {
+      outgoingBloc.add(
+        FtCreateOutgoingTransferEvent.tokenAmountUpdated(amount.decimal),
+      );
+    }
+  }
+
   @override
   Widget build(BuildContext context) => BlocProvider(
-        create: (context) => FtCreateOutgoingTransferBloc(
-          repository: context.read<OutgoingTransferRepository>(),
-          balances: context.read<BalancesBloc>().state.balances,
-          conversionRatesRepository: context.read<ConversionRatesRepository>(),
-          userCurrency: context.read<UserPreferences>().fiatCurrency,
-          transferType: OutgoingTransferType.splitKey,
-          initialToken: widget.token,
-        ),
+        create: (context) => outgoingBloc,
```

Comment:
Never provide the existing entity into `create` method of a provider, check docs why it shouldn't work like that.